guides/source/active_storage_overview.md

```diff
@@ -29,9 +29,10 @@ development and testing and supports mirroring files to subordinate services for
 backups and migrations.
 
 Using Active Storage, an application can transform image uploads with
-[ImageMagick](https://www.imagemagick.org), generate image representations of
-non-image uploads like PDFs and videos, and extract metadata from arbitrary
-files.
+[ImageMagick](https://www.imagemagick.org) or [libvips](https://github.com/libvips/libvips), 
+generate image representations of non-image uploads like PDFs and videos using 
+[poppler](https://poppler.freedesktop.org/) and [ffmpeg](http://ffmpeg.org/), and extract 
+metadata from arbitrary files.
```

Comment:
This sentence is getting a bit cumbersome...  Perhaps parenthesis can help?

```suggestion
(via [ImageMagick](https://www.imagemagick.org) or [libvips](https://github.com/libvips/libvips)), 
generate image representations of non-image uploads like PDFs (via [poppler](https://poppler.freedesktop.org/)) 
and videos (via [ffmpeg](http://ffmpeg.org/)), and extract metadata from arbitrary files.
```


Or perhaps we should just forego mentioning all these tools here?

```suggestion
generate image representations of non-image uploads like PDFs and videos, and extract 
metadata from arbitrary files.
```

Reply:
I'd considered removing all of them. It felt like too much detail about how things are implemented for an opening section, and the "Analyzing Files" section already describes the tools required. 

However, the "Transforming Images" only mentions the image processing gem. Maybe we should add a "requirements" section near the top and mention the necessary gems, libs, etc?

Reply:
There is a note about FFmpeg and Poppler under the [Previewing Files](https://guides.rubyonrails.org/v6.1/active_storage_overview.html#previewing-files) section.  Perhaps, in combination with the other sections, that is sufficient?  Although I'm not opposed to a "Requirements" section either.

Reply:
I'm a bit bothered by the inconsistency. 

Previewing: Big honking red alert about third-party applications needed;
Transforming: A brief mention about adding the `image_processing` gem to the Gemfile but not of the applications;
Analysing: A full mention of all gems and tools, but no mention about adding them to the Gemfile;

I'd rather remove the gems and tools from all three sections above and only mention them in "Requirements" probably as subsection 1.1. I'd also not mention `mini_magick` or `ruby-vips` gems and focus on `image_processing` which already will add both.

```
1.1 Requirements

Various features of Active Storage depend on third party applications which Rails will not install, and you must include separately:

* `ImageMagick` or `libvips` for image analysis and transformations;
* `poppler` or `muPDF` for PDF previews;
* `ffmpeg` for video/audio analysis and video previews;

To enable image transformations the analysis the `image_processing` gem is also necessary:

'''ruby
gem 'image_processing'
'''
```

Reply:
Yes, something like that would be good, though we should be sure to include the warning about licensing implications.  If you'd like we can handle all of that in a follow-up PR, to keep this PR simple.